spec/todoApp.spec.js

```diff
@@ -0,0 +1,177 @@
+const todoApp = require("../src/todoApp");
+
+describe("Todo app", () => {
+  afterEach(function () {
+    todoApp.nextTodoId = 1;
+    todoApp.todoList = [];
+  });
+
+  it("creates a todoItem", () => {
+    // setup
+    const text = "learn TDD";
+    // execute
+    const result = todoApp.create(text);
+    //verify
+    expect(result).toEqual({
+      id: 1,
+      description: text,
+      status: "incomplete",
+    });
+  });
+
+  it("increments IDs by one when adding new item", () => {
+    // setup
+    const task1 = "write tests";
+    const task2 = "write tests";
+    // execute
+    const result1 = todoApp.create(task1);
+    const result2 = todoApp.create(task2);
+    //verify
+    expect(result1).toEqual({
+      id: 1,
+      description: task1,
+      status: "incomplete",
+    });
+    expect(result2).toEqual({
+      id: 2,
+      description: task2,
+      status: "incomplete",
+    });
+  });
+
+  it("returns error message if no description provided", () => {
+    // setup
+    const text = "";
+    // execute
+    const result = todoApp.create(text);
+    //verify
+    expect(result).toEqual("Error: you need to specify the description!!!");
+  });
+
+  it("shows all todo items if the array is not empty", () => {
+    //   it("method showAll() returns array with todos when todos array has at least one item in it", () => {
+    // setup
+    const task1 = "write tests";
+    const todo1 = todoApp.create(task1);
+    const todoList = [todo1];
+    // execute
+    const result = todoApp.showAll();
+    //verify
+    expect(result).toEqual(todoList);
+  });
+
+  it("shows error message if the array is empty", () => {
```

Comment:
This is interesting. In general for technical tests, or in your job, for work that's not in the specification, note it in the Pull Request description or README.md with your reasoning.

Otherwise it's unnecessary work!

Reply:
So in this case, if there are no specifications for different scenarios in the requirements, we shouldn't overthink and overcomplicate our job by creating ones, right?

Reply:
When you don't own the requirements - correct. Instead verify with the owner, or with your team lead 👍